### PR TITLE
Cap for Gradebook points per category 

### LINKED ISSUE #1558: Feature request: the ability to cap a grade category at x points
We currently have the ability to calculate a grade category based on the top n scores.

In contrast, this request is an option where after calculating the total for the category (whether based on the top n or all scores), we calculate the total as the minimum of that score and the number x.

### PR DESCRIPTION
**Pull request by Lyanthropos**
Closes #1558

### CODE REVIEW

**Comment by Lyanthropos** [comment]
@dqnykamp dropped the ball on getting this up, been sitting in my local machine for a week.

File: src/Tools/_framework/ToolPanels/Gradebook.jsx [hunk 1]
```diff
@@ -715,7 +722,12 @@ function GradebookOverview() {
           {
             Header: (
               <div>
-                {`${category} Total`} {description}{' '}
+                {`${category} Total`}{' '}
+                {description.length && (
```
Comment (dqnykamp) [inline src/Tools/_framework/ToolPanels/Gradebook.jsx:726]: This tacks a "0" at the end of categories with no description.

Comment (dqnykamp) [reply src/Tools/_framework/ToolPanels/Gradebook.jsx:726]: Also, I think you can delete description on line 743 when there are no assignments.

File: src/Tools/_framework/ToolPanels/GradebookStudent.jsx [hunk 2]
```diff
@@ -141,18 +137,35 @@ export default function GradebookStudent() {
...
+
+      //Sort by points value and retain the maximumNumber
       scores = scores.sort((a, b) => b - a).slice(0, maximumNumber);
-      let categoryScore = scores.reduce((a, c) => a + c, 0) * scaleFactor;
-      let categoryAssignedPoints = allassignedpoints.reduce((a, c) => a + c, 0) * scaleFactor;
-      if (categoryAssignedPointsAreAllDashes){categoryAssignedPoints = '-'}
 
+      //Scale by scareFactor
+      let scaledScore = scores.reduce((a, c) => a + c, 0) * scaleFactor;
+      let scaledAssignedPoints =
```
Comment (dqnykamp) [inline src/Tools/_framework/ToolPanels/GradebookStudent.jsx:146]: Before summing up allassignedpoints, you first need to sort by points value and retain the maximumNumber as done in 158 for allpossiblepoints.
(This flaw was introduced when assigned points were introduced, but we should correct it now.)

**Review by dqnykamp — CHANGES_REQUESTED** [review]
Need to address changes I mentioned in the other comments.

File: src/Tools/_framework/ToolPanels/GradebookStudent.jsx (hunk 2 above)
Comment (Lyanthropos) [reply src/Tools/_framework/ToolPanels/GradebookStudent.jsx:146]: That is there up above on 142

Comment (dqnykamp) [reply src/Tools/_framework/ToolPanels/GradebookStudent.jsx:146]: Line 142 is for scores.  Need to also do it for scaledAssignPoints

Comment (dqnykamp) [reply src/Tools/_framework/ToolPanels/GradebookStudent.jsx:146]: I.e. before 146, you want to add
```
      //Sort by points value and retain the maximumNumber
      allassignedpoints = allassignedpoints
        .sort((a, b) => b - a)
        .slice(0, maximumNumber);
```

Comment (dqnykamp) [reply src/Tools/_framework/ToolPanels/GradebookStudent.jsx:146]: Otherwise, the assigned points will be too large, as it ignores maximumNumber.

**Pull request merged**